gemini/use-cases/education/resume_generator.ipynb

```diff
@@ -0,0 +1,877 @@
+        "                )  # remove empty strings\n",
+        "                if len(tab_row) > 1 and len(set(tab_row)) == 1:\n",
+        "                    tab_row = tab_row[0]\n",
+        "                    parsed_doc += \"\".join(tab_row) + \"\\n\"\n",
+        "                else:\n",
+        "                    parsed_doc += \", \".join(tab_row) + \"\\n\"\n",
+        "        else:\n",
+        "            parsed_doc += block.text + \"\\n\"  # paragraph text\n",
+        "    return parsed_doc"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "f2541c9f-8d4a-4de5-80af-1d2561ceb3c7"
+      },
+      "outputs": [],
+      "source": [
+        "def parse_resume(res_path: str) -> str:\n",
+        "    \"\"\"\n",
+        "    Driver method to parse resume with fallbacks.\n",
+        "\n",
+        "    Args:\n",
+        "        res_path: Path to the resume file\n",
+        "\n",
+        "    Returns:\n",
+        "        The parsed resume as string\n",
+        "\n",
+        "    Raises:\n",
+        "\n",
+        "    \"\"\"\n",
+        "    file_type = res_path.rsplit(\".\", 1)[-1]\n",
+        "    if file_type == \"docx\":\n",
+        "        parsed_resume = convert_docx_to_text(file_path=res_path).strip()\n",
+        "    elif file_type == \"pdf\":\n",
+        "        parsed_resume = read_pdf(res_path).strip()\n",
+        "    elif file_type == \"txt\":\n",
+        "        with open(res_path) as resume_text:\n",
+        "            parsed_resume = resume_text.read().strip()\n",
+        "    else:\n",
+        "        raise Exception(\"File is not of valid type\")\n",
+        "\n",
+        "    personal_details = check_info(parsed_resume)\n",
+        "    if \"Not Found\" in personal_details.values():\n",
+        "        parsed_resume_2 = docx2txt.process(res_path)\n",
+        "        personal_details_2 = get_info(parsed_resume_2)\n",
+        "        parsed_resume = personal_details_2 + \"\\n\" + parsed_resume\n",
+        "\n",
+        "    return parsed_resume"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "f8963854-36a3-4763-aa11-51fa7ac3733f"
+      },
+      "outputs": [],
+      "source": [
+        "def get_response(\n",
+        "    parsed_res: str,\n",
+        "    parsed_jd: str,\n",
+        ") -> str:\n",
+        "    \"\"\"\n",
+        "    Generate the resume using the chosen model.\n",
+        "\n",
+        "    Args:\n",
+        "        parsed_res: parsed resume as string\n",
+        "        parsed_jd: parsed job description as string\n",
+        "\n",
+        "    Returns:\n",
+        "        Collated resume information as string\n",
+        "    \"\"\"\n",
+        "    model_response_extraction = model.generate_content(\n",
+        "        PROMPT_EXTR % (parsed_res),\n",
```

Comment:
I think it's recommended to use f-strings or `.format()` instead of `%`

Reply:
Replaced with format